### PR TITLE
CEDS-2022 Add Notification & Submission ITs

### PR DESCRIPTION
**Pull request by ed0906**

### CODE REVIEW

File: app/uk/gov/hmrc/exports/movements/services/NotificationService.scala [hunk 1]
```diff
@@ -43,9 +43,6 @@ class NotificationService @Inject()(notificationRepository: NotificationReposito
     }
 
   private def getNotifications(conversationIds: Seq[String]): Future[Seq[NotificationFrontendModel]] =
-    for {
-      notifications <- notificationRepository.findByConversationIds(conversationIds)
-      notificationFrontendModels = notifications.map(NotificationFrontendModel(_))
-    } yield notificationFrontendModels
+    notificationRepository.findByConversationIds(conversationIds).map(_.map(NotificationFrontendModel(_)))
```
Comment (PatrykRudnicki) [inline app/uk/gov/hmrc/exports/movements/services/NotificationService.scala]: 👍

File: test/component/uk/gov/hmrc/exports/movements/NotificationSpec.scala [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+import play.api.libs.json.Json
+import play.api.test.Helpers._
+import uk.gov.hmrc.exports.movements.controllers.routes
+import uk.gov.hmrc.exports.movements.models.notifications.{Entry, Notification, NotificationData, UcrBlock}
+import uk.gov.hmrc.exports.movements.models.submissions.{ActionType, Submission}
+
+/*
+ * Component Tests are Intentionally Explicit with the JSON input, XML & DB output and DONT use TestData helpers.
+ * That way these tests act as a "spec" for our API, and we dont get unintentional API changes as a result of Model/TestData refactors etc.
+ */
```
Comment (PatrykRudnicki) [inline test/component/uk/gov/hmrc/exports/movements/NotificationSpec.scala]: Using scaladoc will make it more visible in the file.
```
/**
  *
  */
```

File: test/component/uk/gov/hmrc/exports/movements/SubmissionSpec.scala [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+import play.api.libs.json.Json
+import play.api.test.Helpers._
+import uk.gov.hmrc.exports.movements.controllers.routes
+import uk.gov.hmrc.exports.movements.models.notifications.{Entry, Notification, NotificationData, UcrBlock}
+import uk.gov.hmrc.exports.movements.models.submissions.{ActionType, Submission}
+
+/*
+ * Component Tests are Intentionally Explicit with the JSON input, XML & DB output and DONT use TestData helpers.
+ * That way these tests act as a "spec" for our API, and we dont get unintentional API changes as a result of Model/TestData refactors etc.
+ */
```
Comment (PatrykRudnicki) [inline test/component/uk/gov/hmrc/exports/movements/SubmissionSpec.scala]: Similar here, comment like above

**Comment by bambuchaAdm** [comment]
Maybe would be worth to mention this scaladoc in README

File: test/component/uk/gov/hmrc/exports/movements/NotificationSpec.scala (hunk 2 above)
Comment (ed0906) [reply test/component/uk/gov/hmrc/exports/movements/NotificationSpec.scala]: Both seem easy to read in my IDE. All the other Component Specs will need changing so I'll leave as is for now

**Pull request merged**